### ISSUE TITLE
Improve --exclude regular expression examples

### ISSUE THREAD

**Issue by nathany** [lycheeverse/lychee#1314]
This is what the documentation has right now:

```
--exclude <EXCLUDE>
          Exclude URLs and mail addresses from checking (supports regex)
```

> You can exclude links from getting checked by specifying regex patterns with `--exclude` (e.g. `--exclude example\.(com|org)`). If a file named .lycheeignore exists in the current working directory, its contents are excluded as well. The file allows you to list multiple regular expressions for exclusion (one pattern per line).

The URL I was attempting to exclude is an unusual case. It's from the generated HTML for a book that is discussing error handling for an invalid web address:

```html
Display the error that occurs when <code>url.Parse</code> is used with an invalid web address, 
such as one containing a space: <code>https://a b.com/</code>
```

The URL appears several times in the file but never as part of an A HREF. The result is:

```
✗ [ERR] https://a/ | Failed: Network error: dns error: no record found for Query { name: Name("a."), query_type: AAAA, query_class: IN }
```

It took me a while to figure out how to ignore this. `--exclude a` would result in **567 Excluded** and variations like `^a` or ^a$` didn't ignore it (based on the example in #1280). From the documentation, it wasn't clear that the solution was to add the scheme.

The solution: `^https://a/$`

Later I found this example in lychee.example.toml, which include the scheme in the regex:

```
# Exclude URLs and mail addresses from checking (supports regex).
exclude = ['^https://www\.linkedin\.com', '^https://web\.archive\.org/web/']
```

Perhaps an additional example or a better example in the README would help?

I don't mind opening a PR for that (not for my specific example, but something like that linkedin example seems good). Before I do, I just wanted to double check that this is working as intended?

**Comment by mre** [comment]
Yes, we can add an additional example. The regex matches on the full URL including the scheme in order to support cases like this. Perhaps we find a way to better describe this behavior. Feel free to open a PR with a suggestion. We could also think about adding more examples to the lychee documentation website, which is also open source.

**Comment by mre** [comment]
I've updated the ["Excluding Links" documentation page](https://lychee.cli.rs/recipes/excluding-links/) with more comprehensive information and real-world examples. Here are the key additions and improvements:

1. Advanced Examples section: Added a variety of practical use cases for link exclusion, such as:
   - Excluding specific file types
   - Filtering social media platforms
   - Checking only links within a specific domain
   - Excluding links with certain URL parameters
   - Excluding links to specific website sections

2. Important Notes section: Added important information about:
   - Full URL matching, including the scheme
   - Precedence of includes over excludes
   - Escaping special regex characters
   - Suggestion to test regex patterns before use

On top of that, I added some tips for effective link exclusion and a short troubleshooting guide.
My hope is that these additions provide users with a better understanding of how to effectively use lychee's exclusion features and help them troubleshoot common issues.

With that, I think we can close this.
